dev/service-registry/service-locator/src/main/scala/com/lightbend/lagom/discovery/ServiceRegistryActor.scala

```diff
@@ -60,27 +63,38 @@ class ServiceRegistryActor @Inject() (unmanagedServices: UnmanagedServices) exte
       import scala.collection.JavaConverters._
       sender() ! RegisteredServices(TreePVector.from(services.asJava))
     case route: Route =>
+      routerFunctions.count(_.isDefinedAt(route)) match {
+        case 0 => logger.warn(s"Can't serve route: $route.")
+        case 1 => // expected case
+        case _ =>
+          val servers = routerFunctions
+            .filter(_.isDefinedAt(route)).map(_.apply(route)).mkString("[", ",", "]")
+          logger.warn(s"Ambiguous route resolution serving route: $route. Route served by $servers")
```

Comment:
It might be nice if it specified explicitly which one is the one it's using, and which ones are masked. Something like "Route served by $servers.head but also matches ${servers.tail.mkString("[", ",", "]")}"